### ISSUE TITLE
STweep throws exception when formatting empty array declaration

### ISSUE THREAD

**Issue by tan-wei-xin-alez** [Barteling/STweep#52]
**Describe the bug**
Formatting the following code with STweep
```
FUNCTION_BLOCK TestPOU
VAR
	arrTest : ARRAY [0..2] OF BOOL := [ ];
END_VAR
```
causes STweep to throw the following exception in CODESYS
```
System.AggregateException: One or more errors occurred.---> System.NullReferenceException:Object reference not set to an instance of an object.
at a.U.A(IParseTree)
at a.U.A(o)
at a.e.A(o)
at a.e.VisitArray_Init(Array_InitContext context)
at STweep.Formatter.CodeParser.ST.STParser.Array_InitContext:.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.A(P,IEnumerable`1 )
at a.e.VisitArray_Spec_Init(Array_Spec_InitContext context)
at STweep.Formatter.CodeParser.ST.STParser.Array_Spec_InitContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitVar_Declaration_Init(Var_Declaration_InitContext context)
at STweep.Formatter.CodeParser.ST.STParser.Var_Declaration_InitContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.VisitChildren(IRuleNode node)
at a.e.VisitChildren(IRuleNode node)
at STweep.Formatter.CodeParser.ST.STParserBaseVisitor`1,VisitVariable_Declaration(Variable_DeclarationContext context)
at STweep.Formatter.CodeParser.ST.STParser.Variable_DeclarationContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitVariable_Declaration_Closed(Variable_Declaration_ClosedContext context)
at STweep.Formatter.CodeParser.ST.STParser.Variable_Declaration_ClosedContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitVariable_Declarations(Variable_DeclarationsContext context)
at STweep.Formatter.CodeParser.ST.STParser.Variable_DeclarationsContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.A(IEnumerable`1 , IParseTree )
at a.e.Visitfunction_Block_Declaration(Function_Block_DeclarationContext context)
at STweep.Formatter.CodeParser.ST.STParser.Function_Block_DeclarationContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitDeclaration(DeclarationContext context)
at STweep.Formatter.CodeParser.ST.STParser.DeclarationContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitProgram(ProgramContext context)
at STweep.Formatter.CodeParser.ST.STParser.ProgramContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at A.r.A(String , CancellationToken )
at A.r.a.A()
at System.Threading.Tasks.Task`1,InnerInvoke()
at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception w.as thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowforNon Success(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.Hand leNon SuccessAndDebuggerNotification(Task task)
at A.r.B.MoveNext()
--- End of inner exception stack trace ---
at System.Threading.Tasks.Task.ThrowlfExceptional(Boolean includeTaskCanceledExceptions)
at System.Threading.Tasks.Task`1,GetResultCore(Boolean waitCompletionNotification)
at System.Threading.Tasks.Task`1,get_Result()
at A.L.A(b)
at A.P.A()
---> (Inner Exception #0) System.NullReferenceException:Object reference not set to an instance of an object.
at a.U.A(IParseTree )
at a.U.A(o)
at a.e.A(o)
at a.e.VisitArray_Init(Array_InitContext context)
at STweep.Formatter.CodeParser.ST.STParser.Array_InitContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.A(P,IEnumerable`1)
at a.e.VisitArray_Spec_Init(Array_Spec_InitContext context)
at STweep.Formatter.CodeParser.ST.STParser.Array_Spec_InitContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitVar_Declaration_Init(Var_Declaration_InitContext context)
at STweep.Formatter.CodeParser.ST.STParser.Var_Declaration_InitContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.VisitChildren(IRuleNode node)
at a.e.VisitChildren(IRuleNode node)
at STweep.Formatter.CodeParser.ST.STParserBaseVisitor`1,VisitVariable_Declaration(Variable_DeclarationContext context)
at STweep.Formatter.CodeParser.ST.STParser.Variable_DeclarationContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitVariable_Declaration_Closed(Variable_Declaration_ClosedContext context)
at STweep.Formatter.CodeParser.ST.STParser.Variable_Declaration_ClosedContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitVariable_Declarations(Variable_DeclarationsContext context)
at STweep.Formatter.CodeParser.ST.STParser.Variable_DeclarationsContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.A(IEnumerable`1 , IParseTree )
at a.e.Visitfunction_Block_Declaration(Function_Block_DeclarationContext context)
at STweep.Formatter.CodeParser.ST.STParser.Function_Block_DeclarationContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitDeclaration(DeclarationContext context)
at STweep.Formatter.CodeParser.ST.STParser.DeclarationContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at a.e.VisitProgram(ProgramContext context)
at STweep.Formatter.CodeParser.ST.STParser.ProgramContext.Accept[TResult](IParseTreeVisitor`1 visitor)
at Antlr4.Runtime.Tree.AbstractParseTreeVisitor`1.Visit(IParseTree tree)
at a.e.Visit(IParseTree tree)
at A.r.A(String , CancellationToken )
at A.r.a.A()
at System.Threading.Tasks.Task`1,InnerInvoke()
at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowforNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at A.r.B.MoveNext()<---
```
<br>

Even utilizing conditional formatting (doesn't matter whether via comments or pragmas) does not prevent error from being thrown as seen below
```
FUNCTION_BLOCK TestPOU
VAR
	{STweep.disable}
	arrTest : ARRAY [0..2] OF BOOL := [ ];
	{STweep.enable}
END_VAR
```

**Expected behavior**
Even if empty arrays will not be supported, conditional formatting should still work so that other sections of the code can be formatted without having to delete the empty array before formatting and adding it in again after formatting

**STweep version info**
STweep formatter for Codesys 3.5.2

**IDE version info**
CODESYS V3.5 SP16 Patch 4 + (32-bit)

**STweep settings**
```
  "Settings": {
    "BlankLines": {
      "BlankLinesAfterMultiLineCase": 0,
      "BlankLinesAfterMultiLineComment": 0,
      "BlankLinesAfterMultiLineDeclaration": 0,
      "BlankLinesAfterMultiLineStatement": 1,
      "BlankLinesAfterSingleLineCase": 0,
      "BlankLinesAfterSingleLineComment": 0,
      "BlankLinesAfterSingleLineDeclaration": 0,
      "BlankLinesAfterSingleLineStatement": 0,
      "BlankLinesAfterStatementWithBody": 1,
      "BlankLinesAfterVarBlocks": 0,
      "BlankLinesAtImplementationFileEnd": 1,
      "BlankLinesBeforeMultiLineComment": 0,
      "BlankLinesBeforeMultiLineDeclaration": 0,
      "BlankLinesBeforeMultiLineStatement": 0,
      "BlankLinesBeforeSingleLineComment": 0,
      "BlankLinesBeforeSingleLineDeclaration": 0,
      "BlankLinesBeforeSingleLineStatement": 0,
      "BlankLinesBeforeStatementWithBody": 1,
      "MaximumBlankLinesToKeepInImplementationCode": 1
    },
    "LineBreaksAndWrapping": {
      "ArrayInitializerParametersLineBreakMode": 0,
      "ChainedBinaryExpressionLineBreakMode": 0,
      "ChainedMethodsLineBreakMode": 0,
      "FunctionBlockInitializerParametersLineBreakMode": 0,
      "InvocationParametersLineBreakMode": 0,
      "KeepExistingLineBreaksInImplementationCode": true,
      "LineBreakAfterLeftBracketInArrayInitializer": false,
      "LineBreakAfterLeftParenthesisInFunctionBlockInitializer": false,
      "LineBreakAfterLeftParenthesisInInvocation": false,
      "LineBreakAfterLeftParenthesisInStructureInitializer": false,
      "LineBreakAfterPragma": 1,
      "LineBreakAfterTypeKeyWord": true,
      "LineBreakBeforeLeftBracketInArrayInitializer": false,
      "LineBreakBeforeLeftParenthesisInFunctionBlockInitializer": false,
      "LineBreakBeforeLeftParenthesisInInvocation": false,
      "LineBreakBeforeLeftParenthesisInStructureInitializer": false,
      "LineBreakBeforeRightBracketInArrayInitializer": false,
      "LineBreakBeforeRightParenthesisInFunctionBlockInitializer": false,
      "LineBreakBeforeRightParenthesisInInvocation": false,
      "LineBreakBeforeRightParenthesisInStructureInitializer": false,
      "LineBreakBeforeStatementsInCase": true,
      "MaxArrayInitializersOnSingleLine": 5,
      "MaxFunctionBlockInitializerParametersOnSingleLine": 5,
      "MaxInvocationParametersOnSingleLine": 5,
      "MaxStructureInitializerParametersOnSingleLine": 5,
      "PlaceStructKeywordOnNewLine": false,
      "PreferWrapBeforeBinaryExpressionOperator": false,
      "StructureInitializerParametersLineBreakMode": 0,
      "WrapBeforeComma": false,
      "WrapLongLinesAtInImplementationCode": 150,
      "WrapLongLinesInImplementationCode": true
    },
    "Spaces": {
      "AfterComma": true,
      "AfterVariableDeclarationColon": true,
      "AroundAdditiveOperator": true,
      "AroundAssignmentOperator": true,
      "AroundAssignmentOperatorInParameterAssignInInvocationCall": false,
      "AroundAssignmentOperatorInParameterAssignInStructureAndFunctionBlockInitializers": false,
      "AroundComment": true,
      "AroundComparisonOperator": true,
      "AroundDot": false,
      "AroundEqualityOperator": true,
      "AroundTermOperator": true,
      "BeforeArrayBrackets": false,
      "BeforeArrayInitializerBrackets": false,
      "BeforeComma": false,
      "BeforeFunctionBlockInitializerParentheses": false,
      "BeforeInvocationParentheses": false,
      "BeforeSemiColon": false,
      "BeforeStructureInitializerParentheses": false,
      "BeforeSubRangeParentheses": false,
      "BeforeVariableDeclarationColon": true,
      "InsideArrayBrackets": false,
      "InsideArrayInitializerBrackets": false,
      "InsideExpressionParentheses": false,
      "InsideFunctionBlockInitializerParentheses": false,
      "InsideInvocationParentheses": false,
      "InsideStructureInitializerParentheses": false,
      "InsideSubRangeParentheses": false,
      "SpacesAroundPragma": true
    },
    "TabsIndentAlignment": {
      "AlignMultilineConstructs": {
        "ArrayInitializers": true,
        "ChainedBinaryExpression": true,
        "FunctionBlockInitializerParameterAssignments": false,
        "FunctionBlockInitializerParameters": true,
        "InvocationCallParameters": true,
        "InvocationCallVariableAssignments": false,
        "MultilineConstructsAlignmentStyleWhenTabsAreUsed": 0,
        "StructureInitializerParameterAssignments": false,
        "StructureInitializerParameters": true
      },
      "AlignSimilarCodeInColumns": {
        "AddressAssignments": true,
        "AlignSimilarCodeInColumnsAlignmentStyleWhenTabsAreUsed": 0,
        "Assignments": false,
        "EndOfLineComments": true,
        "EnumMemberInitializers": true,
        "VariableDeclaration": true,
        "VariableInitializers": true
      },
      "ContinuousLineIndentMultiplier": 1,
      "IndentCasesInCaseStatement": true,
      "IndentDerivedDataTypesInTypeDeclaration": true,
      "IndentElseCaseInCaseStatement": true,
      "IndentSize": 2,
      "IndentStatementsInCase": true,
      "IndentStyle": 0,
      "ParenthesesAlignment": {
        "ArrayInitializerParenthesesStyle": 0,
        "FunctionBlockInitializerParenthesesStyle": 0,
        "FunctionCallParenthesesStyle": 0,
        "StructureInitializerParenthesesStyle": 0
      },
      "TabSize": 2
    }
  }
```

**Comment by Barteling** [comment]
Thanks for reporting!

Is an empty array initializer valid according to the codesys compiler?

**Comment by tan-wei-xin-alez** [comment]
> Thanks for reporting!
> 
> Is an empty array initializer valid according to the codesys compiler?

Yes, apparently it is, I didn't even realize there was an empty array initializer in the code till I ran STweep on it σ(^_^;)

**Comment by Barteling** [comment]
I fixed this in STweep for TwinCAT 4.0.1  which is available here : www.stweep.com/releases

and I fixed this in STweep for Codesys 4.0.0 which will probably be available Monday 31 in the Codesys store.
If you are in a hurry with the STweep for Codesys Version please sent me an email www.stweep.com/contact
